Feature: add `git-flow` commands

asfaltboy opened the pull request:
Add support for the most common commands as simple window commands, as initially requested in #43.

This is still *work in progress*.

What remains to be done:

- [x] Feature/Release command `track`
- ~~Feature commands: `checkout`, `diff` and `rebase`~~ (not really more useful within GitSavvy)
- [ ] Dashboard

About the dashboard, I feel less of a need for it right now. The `branch` dashboard provides an overview of everything I'm looking for, and when I actually need to interact with the branches, running a git command is simple. Also, I'm not really sure how dashboards even work yet.

Closes #43. _(edit)_

divmain commented on core/commands/flow.py:
```diff
@@ -0,0 +1,325 @@
+from ..git_command import GitCommand
+
+
+INIT_REQUIRED_MSG = "Please run `git: flow init` first"
```
Please add `.` at end of string.

divmain commented on Default.sublime-commands:
```diff
@@ -215,5 +215,65 @@
         "command": "gs_reset_reflog"
+    },
+    {
+        "caption": "git flow: init",
```
How would you feel about `flow: init` and so on for the rest of these commands?  The reason I ask is that I'd like not to clutter the command palette for non-users of Git flow.  Particularly with `init` and `pull` in there.

divmain commented on core/commands/flow.py:
```diff
@@ -0,0 +1,325 @@
+                'prefix.support', 'origin', ]
+
+
+class GsFlowCommon(WindowCommand, GitCommand):
```
Add a doc-string here to the class, explaining at a high level what the command does.

divmain commented on docs/flow.md:
```diff
@@ -0,0 +1,29 @@
+# git-flow
```
This should be linked to from `docs/README.md` with a short explanation there, also.

divmain commented:
This looks really clean and straightforward.  Kudos!  A few comments - and I'd love a few more docstrings all around, since I'm not familiar with Git flow, or what each of the commands do.  Would be nice to see that at a glance.

Looks like you'll have one very minor merge conflict.  Resolve that, rebase, and make the changes I requested, and I'll merge this in.  Thanks again, and sorry for the wait.  Haven't had much bandwidth for this lately, but its getting better!

divmain commented:
This closes #43.

asfaltboy commented:
Hey Dale,


Thanks for the review.




I'm abroad for a few weeks with partial internet access; I'll try to get to it in a few days.




I started to work on the dash, but couldn't get over some of the errors. I'll have to leave it for another time.





Best regards,
Pavel Savchenko

On Thu, Jul 16, 2015 at 11:35 AM, Dale Bustad <notifications@github.com>
wrote:

> This closes #43.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/divmain/GitSavvy/pull/214#issuecomment-121877657

asfaltboy commented:
@divmain I followed your comments above, rebased to current master and squashed the commits. Let me know if you see anything else that should be addressed.

divmain commented:
Awesome, thanks!
